connector/protobuf/src/main/scala/org/apache/spark/sql/protobuf/ProtobufDeserializer.scala

```diff
@@ -299,6 +298,36 @@ private[sql] class ProtobufDeserializer(
     }
   }
 
+  private def getFieldValue(record: DynamicMessage, field: FieldDescriptor): AnyRef = {
+    // We return a value if one of:
+    // - the field is repeated
+    // - the field is explicitly present in the serialized proto
+    // - the field is proto2 with a default
+    // - we should materialize the zero value
+    //
+    // Repeated fields have to be treated separately as they cannot have `hasField`
+    // called on them.
+    if (
+      field.isRepeated
+        || record.hasField(field) // checks for the field explicitly being set
+        || field.hasDefaultValue
+        || shouldGetZeroValue(record, field)) {
+      record.getField(field)
+    } else {
+      null
+    }
+  }
+
+  // See the docs for materialize.default.values in [[ProtobufOptions]].
+  // Decides whether we should get the zero values. We do this when
+  // the field is not present in the proto, and theres ambiguity about
+  // whether it was never set or if it was set to its zero value (i.e.
+  // the field lacks field presence information). This is basically
+  // proto3 scalar values only.
+  private def shouldGetZeroValue(record: DynamicMessage, field: FieldDescriptor): Boolean = {
+    !record.hasField(field) && !field.hasPresence && this.materializeZeroValues
```

Comment:
lets move this to `getFieldValue()`.  We can see `!record.hasField(field)` is not needed.

Reply:
okay moved it back